instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/common/client/NettyClientInstrumenterFactory.java

```diff
@@ -51,16 +52,29 @@ public Instrumenter<HttpRequestAndChannel, HttpResponse> createHttpInstrumenter(
 
   public NettyConnectionInstrumenter createConnectionInstrumenter() {
     NettyConnectNetAttributesGetter netAttributesGetter = new NettyConnectNetAttributesGetter();
-    Instrumenter<NettyConnectionRequest, Channel> instrumenter =
+
+    InstrumenterBuilder<NettyConnectionRequest, Channel> instrumenterBuilder =
         Instrumenter.<NettyConnectionRequest, Channel>builder(
                 GlobalOpenTelemetry.get(), instrumentationName, NettyConnectionRequest::spanName)
             .addAttributesExtractor(NetClientAttributesExtractor.create(netAttributesGetter))
             .addAttributesExtractor(PeerServiceAttributesExtractor.create(netAttributesGetter))
-            .setTimeExtractor(new NettyConnectionTimeExtractor())
-            .newInstrumenter(
-                connectionTelemetryEnabled
-                    ? SpanKindExtractor.alwaysInternal()
-                    : SpanKindExtractor.alwaysClient());
+            .setTimeExtractor(new NettyConnectionTimeExtractor());
+    if (!connectionTelemetryEnabled) {
+      // when the connection telemetry is not enabled, netty creates CONNECT spans whenever a
+      // connection error occurs - because there is no HTTP span in that scenario, in raw netty
+      // connection occurs before an HTTP message is even formed
+      // we don't want that span when a higher-level HTTP library (like reactor-netty or async http
+      // client) is used, the connection phase is a part of the HTTP span for these
+      // for that to happen, the CONNECT span will "pretend" to be a full HTTP span when connection
+      // telemetry is off
+      instrumenterBuilder.addAttributesExtractor(MockHttpAttributesExtractor.INSTANCE);
```

Comment:
not sure about `Mock`, maybe instead `Noop`, `Simple`, `Empty`, ...

Reply:
Renamed to `HttpClientSpanKeyAttributesExtractor`